ambry-protocol/src/main/java/com.github.ambry.protocol/MessageInfoListSerde.java

```diff
@@ -31,41 +31,52 @@
   private final List<MessageInfo> messageInfoList;
   static final short MessageInfoListVersion_V1 = 1;
   static final short MessageInfoListVersion_V2 = 2;
+  static final short MessageInfoListVersion_V3 = 3;
 
   private final short version;
 
   private static final byte CRC_PRESENT = (byte) 1;
   private static final byte DELETED = (byte) 1;
 
+  static final int SHORT_FIELD_SIZE_IN_BYTES = 2;
+  static final int INT_FIELD_SIZE_IN_BYTES = 4;
+  static final int LONG_FIELD_SIZE_IN_BYTES = 8;
+
   MessageInfoListSerde(List<MessageInfo> messageInfoList, short version) {
     this.messageInfoList = messageInfoList;
     this.version = version;
   }
 
   int getMessageInfoListSize() {
-    int listcountSize = 4;
     if (messageInfoList == null) {
-      return listcountSize;
+      return INT_FIELD_SIZE_IN_BYTES;
```

Comment:
It seems like the number classes in java 1.8 have BYTES size constants (like Integer.BYTES). We could consider using these here.